great_expectations/data_context/data_context.py

```diff
@@ -3070,18 +3070,17 @@ def add_checkpoint(
             "expectation_suite_ge_cloud_id": expectation_suite_ge_cloud_id,
         }
 
-        batch_data_list = []
-        batch_data = None
+        # DataFrames shouldn't be saved to CheckpointStore
```

Comment:
We shouldn't allow the user to pass a DataFrame into a `RuntimeBatchRequest` and add the DataFrame to the `CheckpointStore`. This would go against the intent of both objects.